dotCMS/src/main/java/com/dotmarketing/portlets/fileassets/ajax/FileAssetAjax.java

```diff
@@ -56,41 +58,50 @@ public FileAssetAjax() {
 	}
 
 
-	public void saveFileText(String contentletInode, String newText, String binField) throws PortalException, SystemException,
-	DotDataException, DotSecurityException, IOException {
-		WebContext ctx = WebContextFactory.get();
-		HttpServletRequest req = ctx.getHttpServletRequest();
-		User user = userAPI.getLoggedInUser(req);
-		boolean respectFrontendRoles = userAPI.isLoggedToFrontend(req);
-		Contentlet cont  = APILocator.getContentletAPI().find(contentletInode, user, respectFrontendRoles);
-		FileAsset fa = APILocator.getFileAssetAPI().fromContentlet(cont);
+	public void saveFileText(final String contentletInode, final String newText, final String binField)
+			throws PortalException, SystemException, DotDataException, DotSecurityException, IOException {
+
+		final WebContext webContext        = WebContextFactory.get();
+		final HttpServletRequest request   = webContext.getHttpServletRequest();
+		final User user 			       = userAPI.getLoggedInUser(request);
+		final boolean respectFrontendRoles = userAPI.isLoggedToFrontend(request);
+		final Contentlet contentlet        = APILocator.getContentletAPI().find(contentletInode, user, respectFrontendRoles);
+		String incomingFileName            = null;
+		String binaryFieldName             = binField;
+
+		if (contentlet.isFileAsset()) {

```

Comment:
I think we should support all binary fields - even if you just have a generic content that has a binary field with text

Reply:
last if support this case, I have tested with a contentlet with a binary as a field